bcda/logging/middleware.go

```diff
@@ -122,8 +124,23 @@ func NewCtxLogger(next http.Handler) http.Handler {
 		if ad, ok := r.Context().Value(auth.AuthDataContextKey).(auth.AuthData); ok {
 			logFields["cms_id"] = ad.CMSID
 		}
+		logFields["transaction_id"] = r.Context().Value(CtxTransactionKey).(string)
 		newLogEntry := &log.StructuredLoggerEntry{Logger: log.API.WithFields(logFields)}
 		r = r.WithContext(context.WithValue(r.Context(), log.CtxLoggerKey, newLogEntry))
 		next.ServeHTTP(w, r)
 	})
 }
+
+// type to create context.Context key
+type CtxTransactionKeyType string
+
+// context.Context key to get the transaction ID from the request context
+const CtxTransactionKey CtxTransactionKeyType = "ctxTransaction"
+
+// Adds a transaction ID to the request context
+func TransactionHandler(next http.Handler) http.Handler {
```

Comment:
Small nit, but might be clearer to name this something like `TransactionID` to match `middleware.RequestID`  or something like `logging.NewTransactionID` to match `logging.NewStructuredLogger`.

Reply:
Yeah, I agree - will update this as well!